lib/resque/plugins/retry.rb

```diff
@@ -453,6 +453,13 @@ def on_failure_retry(exception, *args)
           return
         end
 
+        # If we are "ignoring" the exception, then we decrement the retry
+        # counter, so that the current attempt didn't count toward the retry
+        # counter.
+        if !@ignore_exceptions.nil? && @ignore_exceptions.include?(exception.class)
```

Comment:
Well at that moment we don't know yet what exception would have occurred... the equivalent solution is to move the increment logic "down here" so that the retry counter is incremented only after performing, rather than before.